### PR TITLE
Parties json uploader

### PR DESCRIPTION
**Pull request by gaborh-da**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,9 +42,12 @@ UI_INSTALL_ARTIFACT=ui/node_modules
 $(UI_INSTALL_ARTIFACT): ui/package.json ui/yarn.lock $(JS_CODEGEN_ARTIFACT)
 	cd ui && yarn install --force --frozen-lockfile
 
-.PHONY: build-ui
-build-ui: $(UI_INSTALL_ARTIFACT)
+.PHONY: installui
+installui: $(UI_INSTALL_ARTIFACT)
 
+.PHONY: packui
+packui: installui
```
Comment (richardkapolnai-da) [inline Makefile:49]: we use `-` in names a lot. for consistency, I suggest that: `pack-ui`
or even better: `package`

File: README.md [hunk 2]
```diff
@@ -98,14 +97,15 @@ Supplier
...
 
-6. Run `make build-ui`. Copy `participants.json` into `ui/src` with `cp participants.json ui/src/`. Then run
+6. Run `make installui`. Then run in the `ui` folder (modify the URL, use yours):
 ```
-export REACT_APP_HTTP_BASE_URL=https://wllr6209jx7tgn3g.projectdabl.com
-export REACT_APP_WS_BASE_URL=wss://wllr6209jx7tgn3g.projectdabl.com
+export REACT_APP_HTTP_BASE_URL=https://isfesq2lvxg1y2v8.projectdabl.com
+export REACT_APP_WS_BASE_URL=wss://isfesq2lvxg1y2v8.projectdabl.com
 yarn build
 ```
```
Comment (richardkapolnai-da) [inline README.md:105]: There is something close to this in the Makefile. We should ask the user run simply `make package BASE_URL=isfesq2lvxg1y2v8.projectdabl.com`, and the rest is hidden in the Makefile

File: src/main/daml/DA/RefApps/SupplyChain/Dabl.daml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  SupplyChainParties with
+    buyer = getParty "Buyer"
+    seller = getParty "Seller"
+    supplier = getParty "Supplier"
+    warehouse1 = getParty "Warehouse1"
+    warehouse2 = getParty "Warehouse2"
+    transportCompany1 = getParty "TransportCompany1"
+    transportCompany2 = getParty "TransportCompany2"
+  where
+    getParty name = fromSomeNote ("Party with name " <> name <> " not found.") $ Map.lookup name partyDetails
```
Comment (richardkapolnai-da) [inline src/main/daml/DA/RefApps/SupplyChain/Dabl.daml:30]: please rename it to a non-standard function name (anything not in stdlib)

File: src/main/daml/DA/RefApps/SupplyChain/Dabl.daml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    partyDetails = Map.fromList $ map (\x -> (x.partyName, toParty x.party)) dablParties
+    toParty name = fromSomeNote ("Party cannot be created for " <> name) $ partyFromText name
+
+getSandboxParty : Text -> DablParty
+getSandboxParty name = DablParty name name
+
+fromDablPartiesCanConvert : Script ()
```
Comment (richardkapolnai-da) [inline src/main/daml/DA/RefApps/SupplyChain/Dabl.daml:37]: awesome, please move this under tests

File: ui/src/config.ts [hunk 5]
```diff
@@ -5,35 +5,67 @@
...
 
-export const ledgerId = "supply-chain";
-
 const applicationId = "supply-chain";
 let host = window.location.host.split('.')
 let loginUrl = host.slice(1)
 loginUrl.unshift('login')
-export const dablLoginUrl = loginUrl.join('.') + (window.location.port ? ':' + window.location.port : '') + '/auth/login?ledgerId=' + ledgerId;
 export const isDevMode = process.env.NODE_ENV === 'development'
+export const ledgerId = isDevMode ? "supply-chain" : host[0];
```
Comment (richardkapolnai-da) [inline ui/src/config.ts:21]: couldn't we use `host[0]` always? would `localhost` be a bad ledgerId?

Comment (richardkapolnai-da) [reply ui/src/config.ts:21]: nvm, it maybe something worse like `127.0.0.1`, forget this

Comment (gaborh-da) [reply ui/src/config.ts:21]: I don't think it is a good ledgerId but we could use it.

It's just that we use this in other apps as it is coming from some kind of "template" (Georg's or some other). I suggest not being eclectic in this sense and if we want to change or sort this whole config.ts thing out, do it for all the reference apps (and sorting out would mean much more than just this host, e.g. this app uses different env vars from bond issuance and if I recall correctly one of our other apps has a third solution for the same thing - neither of them considered very comprehensible).

File: src/main/daml/DA/RefApps/SupplyChain/Dabl.daml (hunk 4 above)
Comment (gaborh-da) [reply src/main/daml/DA/RefApps/SupplyChain/Dabl.daml:37]: I would like to copy this file from project to project (as there is no other way) and slightly change it.

You gave me a comment to create a test for it in Bond Issuance where I created this test in a little bit different form in this file.

Do we want to create a specific test file for Dabl.daml that lives in some Test directory for all the projects?

File: README.md (hunk 2 above)
Comment (gaborh-da) [reply README.md:105]: Do you mean `make package REACT_APP_HTTP_BASE_URL=https://isfesq2lvxg1y2v8.projectdabl.com REACT_APP_WS_BASE_URL=wss://isfesq2lvxg1y2v8.projectdabl.com`?

File: src/main/daml/DA/RefApps/SupplyChain/Dabl.daml (hunk 3 above)
Comment (gaborh-da) [reply src/main/daml/DA/RefApps/SupplyChain/Dabl.daml:30]: Is just using `party` as the name okay?

File: README.md (hunk 2 above)
Comment (richardkapolnai-da) [reply README.md:105]: no, I meant what I wrote above, and Makefile could define `REACT_APP_*` variables

File: src/main/daml/DA/RefApps/SupplyChain/Dabl.daml (hunk 3 above)
Comment (richardkapolnai-da) [reply src/main/daml/DA/RefApps/SupplyChain/Dabl.daml:30]: anything reasonal but a std name. so yes, I guess

File: src/main/daml/DA/RefApps/SupplyChain/Dabl.daml (hunk 4 above)
Comment (richardkapolnai-da) [reply src/main/daml/DA/RefApps/SupplyChain/Dabl.daml:37]: Hmmm... ok, keep it, but pls include the prefix `test` in the name.

File: README.md (hunk 2 above)
Comment (gaborh-da) [reply README.md:105]: How can it do that based on `BASE_URL`?

File: src/main/daml/DA/RefApps/SupplyChain/Dabl.daml (hunk 4 above)
Comment (gaborh-da) [reply src/main/daml/DA/RefApps/SupplyChain/Dabl.daml:37]: Note for myself: I have to bring this change to Bond Issuance.

File: src/main/daml/DA/RefApps/SupplyChain/Dabl.daml (hunk 3 above)
Comment (gaborh-da) [reply src/main/daml/DA/RefApps/SupplyChain/Dabl.daml:30]: Note for myself: I have to bring this change to Bond Issuance.

File: src/main/daml/DA/RefApps/SupplyChain/Dabl.daml (hunk 4 above)
Comment (gaborh-da) [reply src/main/daml/DA/RefApps/SupplyChain/Dabl.daml:37]: Also the `map` instead of `<$>`

Comment (gaborh-da) [reply src/main/daml/DA/RefApps/SupplyChain/Dabl.daml:37]: And `dablPartyFromText`

File: Makefile (hunk 1 above)
Comment (gaborh-da) [reply Makefile:49]: I used `package`.

**Review by richardkapolnai-da — APPROVED** [review]

**Pull request merged**